src/repart/repart.c

```
@@ -10353,41 +10357,42 @@ static int vl_method_run(
 
         r = context_ponder(context);
         if (r == -ENOSPC) {
+                uint64_t size = UINT64_MAX, current_size = UINT64_MAX;
+
+                (void) determine_auto_size(context, LOG_DEBUG, &size, &current_size);
```

This is inherited logic, but I think this code is trying to be too smart. The only case when `determine_auto_size` can fail is a partition size would overflow 64 bits. So I think we might just as well treat that as an error.

The issue is that if `determine_auto_size` does fail, the messages below will be bogus, because `need_free` is bogus. I think this should just propagate the error if `determine_auto_size`  fails. @DaanDeMeyer ?

---

For the commit message: expect → except that

---

Should we add another error for the case when the image/disk would overflow 64bit?

---

I don't think that's necessary. It's just another variant of "won't fit".